### ISSUE TITLE
API: Webapi\Rest\Request forces (not intended) additional getter / setter

### ISSUE THREAD

**Issue by shockwavemk** [magento/magento2#3848]
## Steps to reproduce
1. Install Magento from `develop` branch.
2. Create your own API route and a method called “update” 
3. Let method “update” consist of 3 parameters “a” (string), “b” (string) and “o” (an interface for an object)
4. Send a request via REST PUT. Let rest call consist of 2 url-parameters and a json-object sent in body.

Example for rest url: 

```

/rest/V1/example/update/{a}/{b}

```

Example for json-body:  

``` json
{
    "o": {
        "attribute_x": 1,
        "attribute_y": "test"
    }
}
```

The interface for object “o” supports getter and setter for “attribute_x” and “attribute_y”
## Expected result
1. REST request is successful. 
2. Parameters a and b are transferred to method as parameters, Object “o” gets all parameters given in json-body.
## Actual result

Error message appears (for each parameter used in rest route): 

``` json
{

   "message": "Property \"A\" does not have corresponding setter in class \"example\\Module\\Api\\Data\\ObjectInterface\".",
   "trace": "#0 /var/www/html/vendor/magento/framework/Reflection/TypeProcessor.php(591): Magento\\Framework\\Reflection\\TypeProcessor->findAccessorMethodName(Object(Zend\\Code\\Reflection\\ClassReflection), 'A', 'getA', 'isA')\n

}

```

The method 

`overrideRequestBodyIdWithPathParam($urlPathParams)`

in class 

`Magento\Framework\Webapi\Rest\Request`

tries to overwrite / add all parameters from request additionally to json-object if there is only one array in json body. You have to add getter and setter for all parameters used in REST call - this can not be intended.

``` php
if (count($requestBodyParams) == 1 && is_array(end($requestBodyParams))) {
            $requestDataKey = key($requestBodyParams);
            $this->substituteParameters($requestBodyParams[$requestDataKey], $pathParamKey, $pathParamValue);
        }
```

If you add another (useless) parameter to json-body, the problem does not occur:

``` json
{
    "o":{
        "attribute_x": 1,
        "attribute_y": "test"
    },
    "total-useless-param-but-magento-needs-it-to-avoid-error": true
}
```

**Comment by cspruiell** [comment]
Internal ticket MAGETWO-52906 has been created to address this issue.

**Comment by arkadiych** [comment]
@shockwavemk , The issue with overriding parameters in PUT request has been fixed in #3660. 

Please, take notice of a comment from \Magento\Webapi\Controller\Rest\ParamsOverrider::overrideRequestBodyIdWithPathParam:

> ```
>  * If the request body is missing url path parameter as property, it will be added to the body.
>  * This method works only requests with scalar properties at top level or properties of single object embedded
>  * in the request body.
>  * Only the last path parameter value will be substituted from the url in case of multiple parameters.
> ```
